### PR TITLE
Chore: Move admin permissions into redux

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

Moves the permissions required by each part of the app into redux. ~This PR tackles the easy part and the old permissions are not yet removed. In a follow-up I will address the trickier parts to keep both PRs smaller.~

> **Note**
> I think it is the easiest to review this PR commit by commit.

### Why is it needed?

This will pave the path for the runtime EE switch.

#### Learnings

- We should create a menu API/ hook that is capable of handling both cases (settings menu and the main menu) -> Jira Ticket
- We need a better way to setup the global app context/ store via the test utils -> Jira Ticket

### How to test it?

The admin app should work as usually.

### Related issue(s)/PR(s)

- Discussed in https://github.com/strapi/strapi/pull/17023
- Related https://github.com/strapi/strapi/pull/17034

### CODE REVIEW

File: packages/core/admin/admin/src/constants.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        regenerate: [{ action: 'admin::transfer.tokens.regenerate', subject: null }],
+      },
+      'project-settings': {
+        read: [{ action: 'admin::project-settings.read', subject: null }],
+        update: [{ action: 'admin::project-settings.update', subject: null }],
+      },
+    },
+  };
+
+export const ADMIN_PERMISSIONS_SELECTOR = (state) => state.admin_app.permissions;
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/constants.js:92]: Is a selector a constant or a helper/util? 😅 other selectors aren't capitalised is my only reservation – also do we expect to grow `admin_app`? If so should we use `createSelector` or is that overkill? 👀

Comment (gu-stav) [reply packages/core/admin/admin/src/constants.js:92]: > 1. Is a selector a constant or a helper/util?

You are probably right and this should go into a file called `selectors.js`, to follow the other packages.

> 2. also do we expect to grow admin_app

Don't know. Probably. It was there before :)

> 3. If so should we use createSelector or is that overkill? 

Let me explore.

Good remarks!

File: packages/core/admin/admin/src/content-manager/pages/App/index.js [hunk 2]
```diff
@@ -73,6 +72,9 @@ const App = () => {
   const { formatMessage } = useIntl();
   const { startSection } = useGuidedTour();
   const startSectionRef = useRef(startSection);
+  const permissions = useSelector(selectAdminPermissions);
+
+  console.log(permissions);
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/pages/App/index.js:77]: wanna flag this now fyi

Comment (gu-stav) [reply packages/core/admin/admin/src/content-manager/pages/App/index.js:77]: Please leave me alone in here until my tears have dried.

File: packages/core/admin/admin/src/content-manager/pages/CollectionTypeRecursivePath/index.js [hunk 3]
```diff
@@ -16,14 +17,13 @@ import ListViewLayout from '../ListViewLayoutManager';
...
 
-const cmPermissions = permissions.contentManager;
-
 const CollectionTypeRecursivePath = ({
   match: {
     params: { slug },
     url,
   },
 }) => {
+  const permissions = useSelector(selectAdminPermissions);
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/pages/CollectionTypeRecursivePath/index.js:26]: any reason we're not chaining the selector to select exactly what we want?

Comment (gu-stav) [reply packages/core/admin/admin/src/content-manager/pages/CollectionTypeRecursivePath/index.js:26]: Previously every page/ component/ hook had access to all admin permissions. I think it is a good idea to only give them what they need to know about, but if you don't mind I'd like to do this in another iteration.

Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/pages/CollectionTypeRecursivePath/index.js:26]: sgtm

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/ProtectedListView/index.js [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
-  <CheckPagePermissions permissions={adminPermissions.settings['api-tokens'].main}>
-    <ListView />
-  </CheckPagePermissions>
-);
+const ProtectedApiTokenListView = () => {
+  const permissions = useSelector(selectAdminPermissions);
+
+  if (!permissions?.settings?.['api-tokens']?.main) {
+    return null;
+  }
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/ProtectedListView/index.js:14]: this seems odd? we wouldn't render anything and the user would have no feedback – doesn't the `CheckPagePermissions` component handle this?

Comment (gu-stav) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/ProtectedListView/index.js:14]: Good catch. It is leftover and shouldn't be there.

Comment (gu-stav) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/ProtectedListView/index.js]: Removed & rebased.

**Pull request merged**